mountpoint-s3/src/inode/readdir.rs

```diff
@@ -123,6 +127,9 @@ impl ReaddirHandle {
                     warn!("{} has an invalid name and will be unavailable", next.description());
                 } else {
                     let lookup = self.instantiate_remote_inode(next)?;
+                    if inc_lookup_count {
+                        self.inner.remember(&lookup.inode);
+                    }
```

Comment:
So this means non-readdirplus inodes just get deallocated immediately and never get added to the `inodes` dict, right? Probably worth a comment.